pkg/controller/operator/seed/reconciler.go

```diff
@@ -425,7 +425,18 @@ func (r *Reconciler) reconcileConfigMaps(ctx context.Context, cfg *operatorv1alp
 
 	var kubeSystemCreators []reconciling.NamedConfigMapCreatorGetter
 
-	if creator := kubermaticseed.RestoreS3SettingsConfigMapCreator(cfg); creator != nil {
+	// For backward compatibility check both sources for backup and restore configuration.
+	var backupRestore kubermaticv1.SeedBackupRestoreConfiguration
+	if cfg.Spec.SeedController.BackupRestore.Enabled {
```

Comment:
Instead of checking `.Enabled`, can we instead make `BackupRestore` a pointer field and simply check `if cfg.Spec.SeedController.BackupRestore != nil` ? This configuration is optional anyway already.

Reply:
I can remove it from new `SeedBackupRestoreConfiguration` but then I have to return old `KubermaticBackupRestoreConfiguration` type, which has `Enabled` field, for backward comparability.
Because some old configuration might have explicitly set `Enabled=false`.

Reply:
updated with respect to my comment.